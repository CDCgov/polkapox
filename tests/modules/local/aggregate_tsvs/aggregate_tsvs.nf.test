nextflow_process {

    name "Test Process AGGREGATE_TSVS"
    script "modules/local/aggregate_tsvs/aggregate_tsvs.nf"
    process "AGGREGATE_TSVS"

    test("Aggregate TSV Test") {

        when {
            params {

            }
            process {
                """
                input[0] = Channel.of(
                    file("${projectDir}/tests/nf-test-data/test_sub_ivar_summary.txt", checkIfExists: true)
                )
                input[1] = false
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
