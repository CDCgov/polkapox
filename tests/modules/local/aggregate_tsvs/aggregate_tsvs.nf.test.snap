{
    "Aggregate TSV Test": {
        "content": [
            {
                "0": [
                    "all_samples.vcf.summary.txt:md5,a710d645b27c1fc7f8d3b81bf9222ef1"
                ],
                "1": [
                    "versions.yml:md5,96e00fd9a3923d1085e9e698e620e7d5"
                ],
                "txt": [
                    "all_samples.vcf.summary.txt:md5,a710d645b27c1fc7f8d3b81bf9222ef1"
                ],
                "versions": [
                    "versions.yml:md5,96e00fd9a3923d1085e9e698e620e7d5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-11T16:08:30.729510157"
    }
}